src/main/java/filter/Parser.java

```diff
@@ -181,6 +181,8 @@ private FilterExpression parseSeparatedContent(Supplier<FilterExpression> conten
     private FilterExpression parseQualifierContent(QualifierType type) {
         if (type == QualifierType.SORT) {
             return parseSortKeys();
+        } else if (isCompoundId(type, lookAhead())) {
```

Comment:
I realized there is a problem for cases where the `Qualifier` accepts any string as input i.e `title` if we don't check the `QualifierType`.
Take this expression, `title:something/something#`, will result in a filter expression that try to match `(repo:something/something && id:1)` instead of the title